A Content-based Filtering approach for item recommender

Mohamed-Alzahmi opened the pull request:
**The pull request includes:** 
- Content-based Filtering model for item recommendation 
- Documentation files 

**The code is** 
- Tested
- Well-formatted  
- Well-documented

Mohamed-Alzahmi committed:
User recommendations using content_based filtering

content_based filtering approach is developed and tested

Mohamed-Alzahmi committed:
Add documentation files for Grocery Recommender System

Machripe reviewed (commented):
I would like to have a discussion with you regarding these changes. Please check your message in teams.

Mohamed-Alzahmi committed:
Add Content-based model

Add content-based model, add helper functions, separate collaborative from content-based
